### ISSUE TITLE
Merge Accounts sharing the same phone number

### ISSUE THREAD

**Issue by barakd** [dorldor/help-on-the-way#296]
Currently, a user in the system is able to choose how to log in, either as facebook or google.
We are getting more and more complaints about users who are getting confused and log in using the wrong account which gets them to register twice and have two users.

We want to be able to detect those cases and merge the two accounts into one account which is able to login via either login methods.

Current registration flow in the system:
1. User logs in with either facebook or google
2. our server gets a facebook\google token, and based on that token creates a new row in help.private social accounts, and creates a new row in perosn table. see 

- https://github.com/dorldor/help-on-the-way/blob/master/server/src/passport/passportDbQueryRunner.js#L5, and  https://github.com/dorldor/help-on-the-way/blob/master/server/src/passport/providerEndpointsRegisterer.js and https://github.com/dorldor/help-on-the-way/blob/master/server/migrations/sql/V27__get_or_create_user_by_social_function.sql

3. the new user start with the role of volunteer_unverfied
4. the user is being asked to enter his phone number.
5. the user number is being verified by sms. (see https://github.com/dorldor/help-on-the-way/blob/master/server/src/endpoints/phoneVerifier.js)

Basically after step number 5 we should kick in, and decide about a merge stratgey:
My sugestion:
1 .After phone verfication, check if another person in the system of type of volunteer exist.
2. If exists, update help.private scoail account record person id, to point to that perosn id.
3. remove current person
4. return a new JWT session with the "other" person id.

Unclosed point:
What should we do with current user duplicates?
